### ISSUE TITLE
Interesting work! One question for LoRA RL on VLM

### ISSUE THREAD

**Issue by Wangningsen** [ldr0528/Medical-VLM-Fine-tuning-Project#2]
An interesting work. But I still have one question which has bothered me for long time.

I once finetuned with GRPO on Qwen2.5VL 3B/7B with LoRA, and the dataset was also a medical image-text dataset. However, I found out that the training process always collapsed because of unstable reward. Specifically, the reward tended to weave back and forth and didn't seem to increase after many steps. This issue also appears when I switch to a general image-text dataset. After I gave up using LoRA, this issue solved.

I'm very curious about the reward curve of your GRPO training. It will be very nice if you can share any ideas.

**Comment by ldr0528** [comment]
Your questions are very insightful. In fact, my project has run into a very similar issue. In an on-policy, relative-advantage framework like GRPO, if a single update pushes the policy too far away from the reference policy, it can easily lead to:

1.KL divergence spiking (even if you don’t explicitly track KL, it often shows up as the reward oscillating back and forth);

2.Higher variance in the advantage estimates (because the sampling distribution shifts too quickly between iterations);

3.A “weaving” reward curve, or even steadily worsening performance.

Going forward, once I become more proficient, I plan to further refine the project and work out a robust solution to these problems. I look forward to continuing the discussion with you and exchanging ideas in more depth.

**Comment by Wangningsen** [comment]
> Your questions are very insightful. In fact, my project has run into a very similar issue. In an on-policy, relative-advantage framework like GRPO, if a single update pushes the policy too far away from the reference policy, it can easily lead to:
> 
> 1.KL divergence spiking (even if you don’t explicitly track KL, it often shows up as the reward oscillating back and forth);
> 
> 2.Higher variance in the advantage estimates (because the sampling distribution shifts too quickly between iterations);
> 
> 3.A “weaving” reward curve, or even steadily worsening performance.
> 
> Going forward, once I become more proficient, I plan to further refine the project and work out a robust solution to these problems. I look forward to continuing the discussion with you and exchanging ideas in more depth.

This phenomenon only happens when I use **LoRA** for **VLM GRPO**. No similar observation for LLM or full parameter VLM training.  I tracked all important metrics in the training process and they seem to be within normal range. From my perspective this problem is because using LoRA doesn't update enough parameter for GRPO VLM training.
